v2/extensions/dataref_extension_test.go

```diff
@@ -0,0 +1,41 @@
+/*
+ Copyright 2024 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package extensions
+
+import (
+	"testing"
+
+	"github.com/cloudevents/sdk-go/v2/event"
+)
+
+func TestAddDataRefExtensionValidURL(t *testing.T) {
+	e := event.New()
+	expectedDataRef := "https://example.com/data"
+
+	err := AddDataRefExtension(&e, expectedDataRef)
+	if err != nil {
+		t.Fatalf("Failed to add DataRefExtension with valid URL: %s", err)
+	}
+}
+
+func TestAddDataRefExtensionInvalidURL(t *testing.T) {
+	e := event.New()
+	invalidDataRef := "://invalid-url"
+
+	err := AddDataRefExtension(&e, invalidDataRef)
+	if err == nil {
+		t.Fatal("Expected error when adding DataRefExtension with invalid URL, but got none")
+	}
+}
+
+func TestGetDataRefExtensionNotFound(t *testing.T) {
+	e := event.New()
+
+	_, ok := GetDataRefExtension(e)
+	if ok {
+		t.Fatal("Expected not to find DataRefExtension, but did")
+	}
+}
```

Comment:
Looks like Java unit tests :) 

nit: can we use table-driven tests so those tests are easier to modify/add later? Example here: https://github.com/cloudevents/sdk-go/blob/4cc6c2d62d63da00f5b00efd4147e7dd5e40ee4c/v2/protocol/http/protocol_retry_test.go#L25

Reply:
Just noticed you might be following the distributed tracing implementation which uses same style...whatever you prefer

Reply:
good point

Reply:
@matzew I'd like to merge this PR - did you want to update something based on this comment?